Run mypy in CI

**Is your feature request related to a problem? Please describe.**
We're including mypy as a project dependency, but it's not run in CI. Also, there's currently a few issues (on develoment / 93ddb67c85388600ac8c7f347f47c747a9da4310):

```console
$ pipenv run mypy --ignore-missing-imports pyzeebe 
pyzeebe/grpc_internals/zeebe_workflow_adapter.py:15: error: Missing return statement
pyzeebe/grpc_internals/zeebe_workflow_adapter.py:24: error: Missing return statement
pyzeebe/task/task.py:20: error: Incompatible types in assignment (expression has type "None", variable has type "Callable[[Job], Job]")
pyzeebe/worker/task_handler.py:72: error: NotImplemented? not callable
pyzeebe/worker/task_handler.py:79: error: NotImplemented? not callable
pyzeebe/credentials/oauth_credentials.py:40: error: Incompatible return value type (got "None", expected "str")
Found 6 errors in 4 files (checked 33 source files)
```

**Describe the solution you'd like**
Run `pipenv run mypy pyzeebe` in GitHub Actions

**Describe alternatives you've considered**
Pre-commit could also be a nice way to help people lint before pushing

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -1,0 +1,22 @@
+name: Test pyzeebe
+
+on:
+  push:
+  pull_request:
+    branches: [ master, development ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    container: python:3.8
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          pip install pipenv
+          pipenv install --dev
+      - name: Lint with mypy
+        run: |
+          pipenv run mypy pyzeebe

--- a/.mypy.ini
+++ b/.mypy.ini
@@ -1,0 +1,17 @@
+[mypy]
+
+[mypy-grpc]
+ignore_missing_imports = True
+
+[mypy-oauthlib]
+ignore_missing_imports = True
+
+[mypy-requests_oauthlib]
+ignore_missing_imports = True
+
+[mypy-zeebe_grpc.gateway_pb2]
+ignore_missing_imports = True
+
+[mypy-zeebe_grpc.gateway_pb2_grpc]
+ignore_missing_imports = True
+

--- a/pyzeebe/credentials/oauth_credentials.py
+++ b/pyzeebe/credentials/oauth_credentials.py
@@ -37,4 +37,4 @@ class OAuthCredentials(BaseCredentials):
             raise InvalidOAuthCredentialsError(url=url, client_id=client_id, audience=audience)
 
     def get_connection_uri(self) -> str:
-        return None
+        return ""

--- a/pyzeebe/grpc_internals/zeebe_process_adapter.py
+++ b/pyzeebe/grpc_internals/zeebe_process_adapter.py
@@ -16,10 +16,10 @@ class ZeebeProcessAdapter(ZeebeAdapterBase):
             response = self._gateway_stub.CreateProcessInstance(
                 CreateProcessInstanceRequest(bpmnProcessId=bpmn_process_id, version=version,
                                              variables=json.dumps(variables)))
-            return response.processInstanceKey
         except grpc.RpcError as rpc_error:
             self._create_process_errors(
                 rpc_error, bpmn_process_id, version, variables)
+        return response.processInstanceKey
 
     def create_process_instance_with_result(self, bpmn_process_id: str, version: int, variables: Dict,
                                             timeout: int, variables_to_fetch) -> Tuple[int, Dict]:
@@ -29,10 +29,10 @@ class ZeebeProcessAdapter(ZeebeAdapterBase):
                     request=CreateProcessInstanceRequest(bpmnProcessId=bpmn_process_id, version=version,
                                                          variables=json.dumps(variables)),
                     requestTimeout=timeout, fetchVariables=variables_to_fetch))
-            return response.processInstanceKey, json.loads(response.variables)
         except grpc.RpcError as rpc_error:
             self._create_process_errors(
                 rpc_error, bpmn_process_id, version, variables)
+        return response.processInstanceKey, json.loads(response.variables)
 
     def _create_process_errors(self, rpc_error: grpc.RpcError, bpmn_process_id: str, version: int,
                                variables: Dict) -> None:

--- a/pyzeebe/worker/worker.py
+++ b/pyzeebe/worker/worker.py
@@ -124,7 +124,7 @@ class ZeebeWorker(ZeebeTaskRouter):
         return not self.stop_event.is_set() and bool(self._task_threads)
 
     def _watch_task_threads_runner(self, frequency: int = 10) -> None:
-        consecutive_errors = {}
+        consecutive_errors: Dict[str, int] = {}
         while self._should_watch_threads():
             logger.debug("Checking task thread status")
             # converting to list to avoid "RuntimeError: dictionary changed size during iteration"

--- a/tests/unit/grpc_internals/zeebe_process_adapter_test.py
+++ b/tests/unit/grpc_internals/zeebe_process_adapter_test.py
@@ -6,8 +6,9 @@ from uuid import uuid4
 import grpc
 import pytest
 
-from pyzeebe.errors import InvalidJSONError, ProcessDefinitionNotFoundError, ProcessInstanceNotFoundError, ProcessDefinitionHasNoStartEventError, \
-    ProcessInvalidError
+from pyzeebe.errors import InvalidJSONError, ProcessDefinitionNotFoundError, ProcessInstanceNotFoundError, \
+    ProcessDefinitionHasNoStartEventError, \
+    ProcessInvalidError, ZeebeInternalError
 from tests.unit.utils.grpc_utils import GRPCStatusCode
 from tests.unit.utils.random_utils import RANDOM_RANGE
 
@@ -21,18 +22,16 @@ def test_create_process_instance(grpc_servicer, zeebe_adapter):
     assert isinstance(response, int)
 
 
-def test_create_process_instance_common_errors_called(zeebe_adapter):
-    zeebe_adapter._common_zeebe_grpc_errors = MagicMock()
+def test_create_process_instance_raises_grpc_error_correctly(zeebe_adapter):
     error = grpc.RpcError()
     error._state = GRPCStatusCode(grpc.StatusCode.INTERNAL)
 
     zeebe_adapter._gateway_stub.CreateProcessInstance = MagicMock(
         side_effect=error)
 
-    zeebe_adapter.create_process_instance(bpmn_process_id=str(uuid4()), variables={},
-                                          version=randint(0, 10))
-
-    zeebe_adapter._common_zeebe_grpc_errors.assert_called()
+    with pytest.raises(ZeebeInternalError):
+        zeebe_adapter.create_process_instance(bpmn_process_id=str(uuid4()), variables={},
+                                              version=randint(0, 10))
 
 
 def test_create_process_instance_with_result_return_types(grpc_servicer, zeebe_adapter):
@@ -50,19 +49,17 @@ def test_create_process_instance_with_result_return_types(grpc_servicer, zeebe_a
     assert isinstance(response, dict)
 
 
-def test_create_process_instance_with_result_common_errors_called(zeebe_adapter):
-    zeebe_adapter._common_zeebe_grpc_errors = MagicMock()
+def test_create_process_instance_with_result_raises_grpc_error_correctly(zeebe_adapter):
     error = grpc.RpcError()
     error._state = GRPCStatusCode(grpc.StatusCode.INTERNAL)
 
     zeebe_adapter._gateway_stub.CreateProcessInstanceWithResult = MagicMock(
         side_effect=error)
 
-    zeebe_adapter.create_process_instance_with_result(bpmn_process_id=str(uuid4()), variables={},
-                                                      version=randint(0, 10), timeout=0,
-                                                      variables_to_fetch=[])
-
-    zeebe_adapter._common_zeebe_grpc_errors.assert_called()
+    with pytest.raises(ZeebeInternalError):
+        zeebe_adapter.create_process_instance_with_result(bpmn_process_id=str(uuid4()), variables={},
+                                                          version=randint(0, 10), timeout=0,
+                                                          variables_to_fetch=[])
 
 
 def test_cancel_process(grpc_servicer, zeebe_adapter):

